charts/backstage/templates/_helpers.tpl

```diff
@@ -57,8 +59,14 @@ Return the Postgres Database Secret Name
 Return the Postgres databaseSecret key to retrieve credentials for database
 */}}
 {{- define "backstage.postgresql.databaseSecretKey" -}}
-{{- if .Values.postgresql.auth.existingSecret -}}
-    {{- .Values.postgresql.auth.secretKeys.userPasswordKey  -}}
+{{- if (or ((((.Values).global).postgresql).auth).existingSecret .Values.postgresql.auth.existingSecret) }}
```

Comment:
What about adding the following immediately below the `define` statement to initialize a variable for the default Database Secret Key

```
{{- $defaultDatabaseSecretKey := "password" -}}
```

then reference it in the `else` clauses to avoid duplicate hardcoded properties

```
{{- print $defaultDatabaseSecretKey -}}
```

Reply:
Good call. Done 🙂